### DISCUSSION TITLE
weird number of es hits

### DISCUSSION THREAD

**Discussion by 1998729** [jertel/elastalert2#1067] [Q&A] [answered]
docker image version: 2.3.0
es cluster version: 6.8 

I am currently encountering a strange phenomenon. I once created a query rule, and then temporarily deleted the rule file for some reasons. Later, I created this rule in the rule directory, but the first query has an astonishing error Number of queries: 11825 hits and match,

Can anyone tell me how this amazing number is caused, the normal number is controlled within 20

**Comment by 1998729** [comment]
Phenomena data:

<img width="1183" alt="image" src="https://user-images.githubusercontent.com/88186802/209773880-5b4ffc41-6951-40af-b426-2591eeacf5dd.png">

**Comment by 1998729** [comment]
rule_file:

type: frequency
index: ee-sla-base-%Y.%m.%d
use_strftime_index: true
run_every:
  minutes: 1
buffer_time:
  minutes: 3
timeframe:
  minutes: 1
realert:
  minutes: 3
num_events: 1
filter:
- query:
    query_string:
      query: 'business: "base" AND http_host: "is-gateway." AND whitelist.keyword:
        0 NOT uri: websocket NOT uri: ws NOT uri: socket.io AND status: >=500'
query_key:
- status
- uri

**Comment by 1998729** [comment]
This kind of sudden large number of hits, from which aspects can we generally query? Because we used kibana and grafana to check the data at that time, the maximum number was within 20.

**Comment by 1998729** [comment]
alert_text_args:
- http_host
- uri
- status
- num_matches
alert_subject_args:
- http_host
- uri
- status
- num_matches
alert_text_type: alert_text_only
alert_text: '{0} {1} {2}: {3}' <-------


result:
is-gateway..com /api 500: 11825

**Comment by 1998729** [comment]
Is it possible that it is a bug? I deleted the rule file, but it is still calculating in the memory, and the job has not been removed. At this time, I added the file back, and it aggregated all the previous data.

**Comment by 1998729** [comment]
@jertel  help 😭

**Reply by nsano-rururu** [reply]
https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#disabling-a-rule

**Reply by 1998729** [reply]
> https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#disabling-a-rule

<img width="405" alt="image" src="https://user-images.githubusercontent.com/88186802/209804761-f2c76c0c-0495-4a0d-8086-59b289042e88.png">
The service is not stopped, but the previous rule file is removed, and then a new rule is added after a while, the rulename is the same

**Reply by 1998729** [reply]
> https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#disabling-a-rule

Are you saying that if you want to remove the rule file, you need to add a parameter in the file first: is_enabled=false?

**Reply by nsano-rururu** [reply]
Another option is to restart elastalert2

**Reply by nsano-rururu** [reply]
https://github.com/Yelp/elastalert/issues/1677

**Reply by nsano-rururu** [reply]
https://github.com/Yelp/elastalert/issues/2893

**Reply by nsano-rururu** [reply]
https://github.com/johnsusek/praeco/issues/165#issuecomment-643801295

**Reply by 1998729** [reply]
oh my god! It seems that I have no way to manage the rule directory in an automated way. My current practice is to disable rules = delete files, enable rules = create files.

**Reply by nsano-rururu** [reply]
The pull request below has been merged into elastalert2
https://github.com/Yelp/elastalert/pull/3036

**Reply by nsano-rururu** [reply]
https://github.com/Yelp/elastalert/issues/2043

**Reply by 1998729** [reply]
A new discovery, I found that after I enabled this rule, he started to query from the event I disabled last time?

**Reply by 1998729** [reply]
![image](https://user-images.githubusercontent.com/88186802/209895064-d4b894ac-864f-4fdc-9390-d03ab73cb2c2.png)

+8 hour

**Reply by 1998729** [reply]
![image](https://user-images.githubusercontent.com/88186802/209895620-b80224ca-dd46-4c32-a670-b738cd667edb.png)
In this case, won't the data between the end of the last query and the present be queried and aggregated at the next startup?

**Reply by 1998729** [reply]
![image](https://user-images.githubusercontent.com/88186802/209896135-08302b86-e13f-4889-9ef9-fa7a9deb179b.png)

**Reply by 1998729** [reply]
@nsano-rururu
I verified again to use is_enable to do this, but the result is the same. For example, if it is disabled for 2 hours, then when it is enabled, the query matching will still be counted from 2 hours ago, and an amazing hit will be aggregated, and then an alarm will be sent!

**Reply by 1998729** [reply]
![image](https://user-images.githubusercontent.com/88186802/209897069-8bf5af67-2910-4204-9f74-b0eb3e2bc177.png)

**Reply by 1998729** [reply]
It seems that the elastalert index records the time of the last query, and after a long time, after enabling this rule, it will query at one time, and then aggregate and send an alarm! How to solve this problem?

**Reply by jertel** [reply]
You will need to use unique rule names if you want re-enable a previously disabled rule, but without any history.

Ex:
```
myrule_20221229125959
```

**Reply by 1998729** [reply]
> You will need to use unique rule names if you want re-enable a previously disabled rule, but without any history.
> 
> Ex:
> 
> ```
> myrule_20221229125959
> ```

Considering the name of the rule generated by the platform, I decided to use an extreme method: when I disable or enable it from the platform dimension, I will delete the disabled rule from the elastalert metadata index!

**Comment by kkbit96** [comment]
Why am i getting these emails?

**Reply by jertel** [reply]
GitHub suffered a large spam bot attack starting yesterday around 6pm EST. There were several posts to most popular GitHub projects, where the spammer @ mentioned random GitHub users to try and get them to click on the spam link. If you're receiving messages from this thread, it's because of this spam attack. The offending messages have already been removed from the discussion thread.

**Comment by 1998729** [comment]
disable or delete action
![image](https://user-images.githubusercontent.com/88186802/210028243-7fe7df0a-62e5-4502-815d-9e857e35fe11.png)

 curl -X POST -H 'Content-Type: application/json' http://127.0.0.1:9523/elastalert_status/_delete_by_query  -d '{"query": {"term": {"rule_name": "xxxx"}}}'

**Reply by 1998729** [reply]
Obviously, this approach failed, he may have written metadata into memory! ! !

**Comment by 1998729** [comment, marked as answer]
According to the timestamp, as long as it is disabled and then enabled, the current timestamp will be spliced in the rule_name field of the current rule
